servicetest/filesystem/test_filesystem_strace.py

```diff
@@ -0,0 +1,191 @@
+        if re.search('^mkdir', line):
+            p = re.compile('^mkdir\(\"(.*)\",')
+            files.append(p.match(line).group(1))
+
+    return files
+
+@pytest.mark.usefixtures("create_testoutput_dir", "dbus_launch", "agent", "assert_no_proxy")
+class TestFileSystemStrace(object):
+    """ This suite does some strace testing of the creation and deletion of
+    containers and make sure we don't create or modify files outside the areas
+    we should be working within.
+
+    The tests are performed by prefixing the softwarecontainer-agent binary
+    with strace and logging the output into a logfile and then parsing the
+    logfile for the files and directories being created. We can then check the
+    paths and make sure they are valid.
+    """
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_bindmount_sockets_strace(self):
+        """ Test that sockets are bindmountable as expected.
+        """
+        startword = "======STARTSTARTSTART====="
+        endword = "======ENDENDEND====="
+
+        with open(logfile_path(), mode="a") as f:
+            f.write("\n" + startword + "\n")
+
+        absolute_test_file = os.path.join("/tmp/", TESTFILE)
+
+        if not os.path.exists(absolute_test_file):
+            server = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            server.bind(absolute_test_file)
+
+        ca = Container()
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true}]'
+
+        try:
+            ca.start(DATA)
+            ca.bindmount(absolute_test_file, "/gateways/testfile", True)
+            # Give the command time to run inside the container
+            time.sleep(0.5)
+            assert os.path.exists(absolute_test_file) is True
+            ca.launch_command('{}/fileapp.py check {}'
+                              .format(ca.get_bind_dir(), "/gateways/testfile"))
+            time.sleep(0.5)
+            assert os.path.exists(absolute_test_file) is True
+        finally:
+            ca.terminate()
+
+        server.close()
+
+        if os.path.exists(absolute_test_file):
+            os.remove(absolute_test_file)
+
+        time.sleep(0.5)
+        with open(logfile_path(), mode="a") as f:
+            f.write("\n" + endword + "\n")
+
+        # Get the logblock from the above run containing strace logs from
+        # SoftwareContainerAgent
+        buf = get_logblock(logfile_path(), startword, endword)
+
+        # Read out the created files and directories from the caught buffer
+        files = get_strace_created_files(buf)
+
+        # Directory patterns that are acceptable to create inside.
+        patterns = ['/tmp/container/SC-0',
+                    '/usr/var/lib/lxc/SC-0',
+                    '/sys/fs/cgroup/',
+                    '/run/lxc/']
+
+        # Find all instances in files matching any of the patterns and create
+        # a fileset containing all "OK" files
+        fileset = list()
+        for file in set(files):
```

Comment:
`file` is a predefined function in python 2.x and should not be used as a variable name.
https://docs.python.org/2.7/library/functions.html#file